lib/iris/tests/__init__.py

```diff
@@ -143,7 +147,8 @@
     plt.switch_backend('tkagg')
     _DISPLAY_FIGURES = True
 
-_DEFAULT_IMAGE_TOLERANCE = 10.0
+# Threading non re-entrant blocking lock to ensure thread-safe plotting.
+_lock = threading.Lock()
```

Comment:
A **non re-entrant** lock is required here to ensure that there is no cross-pollination between tests using the wrong plot figures or axes.